### DISCUSSION TITLE
Rectangle config

### DISCUSSION THREAD

**Discussion by MustCodeAl** [asmvik/yabai#2202] [General]
is there any yabairc configurations that mimimic the entire rectangles hotkeys
![shortcutScreenshot](https://github.com/koekeishiya/yabai/assets/87888006/c3058290-d19f-4ffa-a9ac-01dd897bcd60)

**Comment by asmvik** [comment]
I'm not aware of any pre-created configs for this, but just quickly looking at the screenshot pretty much all of it should be doable with `window --grid` command.

**Comment by globalshrug** [comment]
> is there any yabairc configurations that mimimic the entire rectangles hotkeys

I have a very basic config that includes some of what you're looking for. You'll need to tweak it to fit your needs.

```
# If the focused window is in float mode.
# Fill screen
ctrl + alt - f :  yabai -m window --grid 1:1:0:0:1:1
# Left half
ctrl + alt - a :  yabai -m window --grid 1:2:0:0:1:1
# Right half
ctrl + alt - d :  yabai -m window --grid 1:2:1:0:1:1
# Top half
ctrl + alt - w :  yabai -m window --grid 2:1:0:0:1:1
# Bottom half
ctrl + alt - s :  yabai -m window --grid 2:1:0:1:1:1
# Top left
ctrl + alt - q :  yabai -m window --grid 2:2:0:0:1:1
# Bottom left
ctrl + alt - z :  yabai -m window --grid 2:2:0:1:1:1
# Top right
ctrl + alt - e :  yabai -m window --grid 2:2:1:0:1:1
# Bottom right
ctrl + alt - c :  yabai -m window --grid 2:2:1:1:1:1
# Center
ctrl + alt - x :  yabai -m window --grid 3:3:1:1:1:1
```
